### PR TITLE
Add option to duplicate columns

### PR DESCRIPTION
**Pull request by aptkingston**
## Description
This PR adds the ability to duplicate columns in grids. The name is automatically generated, and will increment as required. All column metadata is duplicated.

image
image

Working with options, copying over metadata successfully:
image

Automatic incrementing of names when duplicating again:
image

Addresses: 
- https://linear.app/budibase/issue/BUDI-7418/duplicate-a-column-in-data-section

### CODE REVIEW

**Comment by aptkingston** [comment]
Await release before merging.

File: packages/frontend-core/src/components/grid/cells/HeaderCell.svelte [hunk 1]
```diff
@@ -118,6 +120,33 @@
...
   }
 
+  const duplicateColumn = async () => {
+    open = false
+
+    // Generate new name
+    let newName = `${column.name} copy`
+    let attempts = 2
+    while ($definition.schema[newName]) {
+      newName = `${column.name} copy ${attempts++}`
```
Comment (melohagan) [inline packages/frontend-core/src/components/grid/cells/HeaderCell.svelte:130]: What happens when it runs out of attempts, does it just throw an error?

File: packages/frontend-core/src/components/grid/cells/HeaderCell.svelte [hunk 2]
```diff
@@ -118,6 +120,33 @@
...
+    const existingColumnDefinition = $definition.schema[column.name]
+    await datasource.actions.saveDefinition({
+      ...$definition,
+      schema: {
+        ...$definition.schema,
+        [newName]: {
+          ...existingColumnDefinition,
+          name: newName,
+          schema: {
+            ...existingColumnDefinition.schema,
```
Comment (melohagan) [inline packages/frontend-core/src/components/grid/cells/HeaderCell.svelte:143]: If we're already spreading the `existingColumnDefinition` on line 140, so we need specify the schema property here?

**Review by melohagan — APPROVED** [review]
LGTM - handy feature!

File: packages/frontend-core/src/components/grid/cells/HeaderCell.svelte (hunk 1 above)
Comment (aptkingston) [reply packages/frontend-core/src/components/grid/cells/HeaderCell.svelte:130]: Attempts increments, so it never runs out!

File: packages/frontend-core/src/components/grid/cells/HeaderCell.svelte (hunk 2 above)
Comment (aptkingston) [reply packages/frontend-core/src/components/grid/cells/HeaderCell.svelte:143]: We need to change the schema from the existing column definition, so this is how you copy and change a nested property without mutating the base! If I didn't spread the base schema then schema would only have whatever properties specified here.

File: packages/frontend-core/src/components/grid/cells/HeaderCell.svelte (hunk 1 above)
Comment (melohagan) [reply packages/frontend-core/src/components/grid/cells/HeaderCell.svelte:130]: Ah yes, nice!

**Pull request merged**